common-libs/common/src/Affinity.ts

```diff
@@ -18,12 +19,46 @@ if (!fetch) {
 export class Affinity {
   private readonly _apiKey: string
   private readonly _registryUrl: string
+  private readonly _metricsUrl: string
+  private readonly _metricsService: any
   private readonly _digestService: any
+  protected _component: EventComponent
 
   constructor(options: AffinityOptions = {}) {
     this._apiKey = options.apiKey
     this._registryUrl = options.registryUrl || DEFAULT_REGISTRY_URL
+    this._metricsUrl = options.metricsUrl || DEFAULT_METRICS_URL
+    this._component = options.component
```

Comment:
@everdark what will be if component will be not passed? I believe we should use some default one for such cases

Reply:
Agreed.